conductor_api/src/conductor/base.rs

```diff
@@ -172,6 +179,52 @@ impl Conductor {
         self.config.clone()
     }
 
+    /// Starts a new thread which monitors each instance's signal channel and pushes signals out
+    /// all interfaces the according instance is part of.
+    /// TODO: add kill-switch to this thread to close it on shutdown / also make sure to recreate if interfaces get changed..
```

Comment:
@lucksus maybe this part is important? You've been looking at the code more recently, so you might have a better sense. Do we need to do anything extra to handle the case when interfaces get restarted?

Reply:
I've added the kill-switch which gets triggered on shutdown. Also added shutdown to the Drop implementation.